### ISSUE TITLE
Make the mccabe checker ignore the case of a single top-level elif chain or match statement

### ISSUE THREAD

**Issue by saolof** [charliermarsh/ruff#3597]
This is a somewhat common edge case where there shouldn't be a need to use noqa comments. Non-looping functions with a nesting level of 1 should be excluded from the mccabe checker, or there should be a separate error code for this particular kind of pattern if you want to lint or ignore it separately, such as "elif chain exceeds mccabe complexity".

**Comment by antonagestam** [comment]
I'd like to work on this.

**Comment by charliermarsh** [comment]
Confirming my understanding: are you describing something akin to a top-level switch statement? Like:

```py
if foo == "bar":
  return 1
elif foo == "baz:
  return 2:
else ...
```

**Comment by saolof** [comment]
Yes, exactly. There is a clear difference in complexity between a switch statement with 10 cases, and a piece of code that has 3 nested for loops with if-clauses randomly distributed inside.

**Comment by antonagestam** [comment]
@saolof I agree, and I think it can sometimes be detrimental to try and break apart code like this. The result isn't likely to be easier for a human to reason about, and is likely to lower coherence. Single-level structures are the exception to the rule, because they are already easy to reason about and follow.

**Comment by antonagestam** [comment]
Sorry, I haven't had much time to spend on this. I might get to it at some point, but if someone else is keen to work on this, please go ahead.
